src/MLOps.NET/Storage/Repositories/RunRepository.cs

```diff
@@ -169,5 +175,23 @@ public RegisteredModel GetLatestRegisteredModel(Guid experimentId)
 
             return this.registeredModelResolver.BuildEntity(db, latestRegisteredModel);
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="TInput"></typeparam>
+        /// <typeparam name="TOutput"></typeparam>
+        /// <param name="runId"></param>
+        public async Task RegisterSchema<TInput, TOutput>(Guid runId) where TInput : class where TOutput : class
```

Comment:
Let's do the schema resolution outside of the `RunRepository`. I want to try to keep the `RunRepository` focused on only managing the DB context. We can call the `SchemaGenerator` in e.g. the `LifeCycleCatalog` and pass the result in here.